yt_dlp/extractor/peloton.py

```diff
@@ -0,0 +1,336 @@
+                                    }).encode(),
+                                    headers={
+                                        'Content-Type': 'application/json'
+                                    })
+            except ExtractorError as e:
+                if isinstance(e.cause, compat_HTTPError) and e.cause.code == 401:
+                    json_string = self._webpage_read_content(e.cause, None, video_id)
+                    res = self._parse_json(json_string, video_id)
+                    if res['message'] == 'Login failed':
+                        raise ExtractorError(res['message'], expected=True)
+                    else:
+                        raise ExtractorError(res['message'])
+                else:
+                    raise
+        else:
+            self.raise_login_required()
+
+    def _get_ride(self, video_id):
+        return self._download_json(self._RIDE_URL_TEMPLATE % video_id, video_id)
+
+    def _get_token(self, video_id):
+        try:
+            subscription = self._download_json(self._SUBSCRIPTION_URL, video_id,
+                                               note='Downloading token',
+                                               data=json.dumps({}).encode(),
+                                               headers={
+                                                   'Content-Type': 'application/json'
+                                               })
+        except ExtractorError as e:
+            if isinstance(e.cause, compat_HTTPError) and e.cause.code == 403:
+                json_string = self._webpage_read_content(e.cause, None, video_id)
+                res = self._parse_json(json_string, video_id)
+                if res['message'] == 'Stream limit reached':
+                    raise ExtractorError(res['message'], expected=True)
+                else:
+                    raise ExtractorError(res['message'])
+            else:
+                raise
+        return subscription['token']
+
+    def _extract(self, video_id):
+        try:
+            self._start_session(video_id)
+        except ExtractorError as e:
+            if isinstance(e.cause, compat_HTTPError) and e.cause.code == 401:
+                self._login(video_id)
+                self._start_session(video_id)
+            else:
+                raise
+
+        metadata = self._get_ride(video_id)
+        if not metadata.get('ride'):
+            raise ExtractorError('Missing stream metadata')
+
+        token = self._get_token(video_id)
+
+        is_live = False
+        subtitles = {}
+        content_format = try_get(metadata, lambda x: x['ride']['content_format'], compat_str) or 'video'
+        if content_format == 'audio':
+            url = self._MANIFEST_URL_TEMPLATE % (metadata['ride'].get('vod_stream_url'), compat_urllib_parse.quote(token))
+            formats = [{
+                'url': url,
+                'ext': 'm4a',
+                'format_id': 'audio',
+                'resolution': 'audio only'
```

Comment:
`resolution` will be automatically set later. What you need to set is `'vcodec': 'none'` to signify that the format is audio only